### DISCUSSION TITLE
error on starting server with plugins

### DISCUSSION THREAD

**Discussion by RSM-SBA** [appium/appium#15818] [Q&A] [answered]
I tried now the whole day but i have no clue why this error occurs.
Anyone an idea why the appium-support could not be found ?

Simply done :
npm install appium@next --chromedriver-cdnurl=http://chromedriver.storage.googleapis.com --drivers=uiautomator2 --plugins=images

and afterwards : 
node_modules/.bin/appium --plugins images

**console log :**
npm install appium@next --chromedriver-cdnurl=http://chromedriver.storage.googleapis.com --drivers=uiautomator2 --plugins=images

> appium@2.0.0-beta.13 postinstall /Users/jenkins/HomeOfQA/workspace/RMA-Android-Smoke-Java/uitesting/node_modules/appium
> node ./postinstall.js

Installing the driver uiautomator2...
Installing the plugin images...
npm WARN uitesting No description

+ appium@2.0.0-beta.13
added 425 packages from 518 contributors and audited 426 packages in 34.777s

34 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

❯ node_modules/.bin/appium --plugins images
[Appium] Attempting to load plugin images...
[Appium] Could not load plugin 'images', so it will not be available. Error in loading the plugin was: Cannot find module 'appium-support'
[Appium] Require stack:
[Appium] - /Users/jenkins/.appium/@appium/images-plugin/node_modules/@appium/base-driver/build/lib/protocol/protocol.js
[Appium] - /Users/jenkins/.appium/@appium/images-plugin/node_modules/@appium/base-driver/build/lib/protocol/index.js
[Appium] - /Users/jenkins/.appium/@appium/images-plugin/node_modules/@appium/base-driver/build/lib/basedriver/driver.js
[Appium] - /Users/jenkins/.appium/@appium/images-plugin/node_modules/@appium/base-driver/build/index.js
[Appium] - /Users/jenkins/.appium/@appium/images-plugin/node_modules/@appium/images-plugin/build/lib/plugin.js
[Appium] - /Users/jenkins/.appium/@appium/images-plugin/node_modules/@appium/images-plugin/build/index.js
[Appium] - /Users/jenkins/HomeOfQA/workspace/RMA-Android-Smoke-Java/uitesting/node_modules/appium/build/lib/extension-config.js
[Appium] - /Users/jenkins/HomeOfQA/workspace/RMA-Android-Smoke-Java/uitesting/node_modules/appium/build/lib/cli/parser-helpers.js
[Appium] - /Users/jenkins/HomeOfQA/workspace/RMA-Android-Smoke-Java/uitesting/node_modules/appium/build/lib/cli/args.js
[Appium] - /Users/jenkins/HomeOfQA/workspace/RMA-Android-Smoke-Java/uitesting/node_modules/appium/build/lib/cli/parser.js
[Appium] - /Users/jenkins/HomeOfQA/workspace/RMA-Android-Smoke-Java/uitesting/node_modules/appium/build/lib/main.js
[debug] [Appium] Error: Cannot find module 'appium-support'
[debug] [Appium] Require stack:
[debug] [Appium] - /Users/jenkins/.appium/@appium/images-plugin/node_modules/@appium/base-driver/build/lib/protocol/protocol.js
[debug] [Appium] - /Users/jenkins/.appium/@appium/images-plugin/node_modules/@appium/base-driver/build/lib/protocol/index.js
[debug] [Appium] - /Users/jenkins/.appium/@appium/images-plugin/node_modules/@appium/base-driver/build/lib/basedriver/driver.js
[debug] [Appium] - /Users/jenkins/.appium/@appium/images-plugin/node_modules/@appium/base-driver/build/index.js
[debug] [Appium] - /Users/jenkins/.appium/@appium/images-plugin/node_modules/@appium/images-plugin/build/lib/plugin.js
[debug] [Appium] - /Users/jenkins/.appium/@appium/images-plugin/node_modules/@appium/images-plugin/build/index.js
[debug] [Appium] - /Users/jenkins/HomeOfQA/workspace/RMA-Android-Smoke-Java/uitesting/node_modules/appium/build/lib/extension-config.js
[debug] [Appium] - /Users/jenkins/HomeOfQA/workspace/RMA-Android-Smoke-Java/uitesting/node_modules/appium/build/lib/cli/parser-helpers.js
[debug] [Appium] - /Users/jenkins/HomeOfQA/workspace/RMA-Android-Smoke-Java/uitesting/node_modules/appium/build/lib/cli/args.js
[debug] [Appium] - /Users/jenkins/HomeOfQA/workspace/RMA-Android-Smoke-Java/uitesting/node_modules/appium/build/lib/cli/parser.js
[debug] [Appium] - /Users/jenkins/HomeOfQA/workspace/RMA-Android-Smoke-Java/uitesting/node_modules/appium/build/lib/main.js
[debug] [Appium]     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:815:15)
[debug] [Appium]     at Function.Module._load (internal/modules/cjs/loader.js:667:27)
[debug] [Appium]     at Module.require (internal/modules/cjs/loader.js:887:19)
[debug] [Appium]     at require (internal/modules/cjs/helpers.js:74:18)
[debug] [Appium]     at Object.<anonymous> (/Users/jenkins/.appium/@appium/images-plugin/node_modules/@appium/base-driver/lib/protocol/protocol.js:2:1)
[debug] [Appium]     at Module._compile (internal/modules/cjs/loader.js:999:30)
[debug] [Appium]     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
[debug] [Appium]     at Module.load (internal/modules/cjs/loader.js:863:32)
[debug] [Appium]     at Function.Module._load (internal/modules/cjs/loader.js:708:14)
[debug] [Appium]     at Module.require (internal/modules/cjs/loader.js:887:19)
[debug] [Appium]     at require (internal/modules/cjs/helpers.js:74:18)
[debug] [Appium]     at Object.<anonymous> (/Users/jenkins/.appium/@appium/images-plugin/node_modules/@appium/base-driver/lib/protocol/index.js:3:1)
[debug] [Appium]     at Module._compile (internal/modules/cjs/loader.js:999:30)
[debug] [Appium]     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
[debug] [Appium]     at Module.load (internal/modules/cjs/loader.js:863:32)
[debug] [Appium]     at Function.Module._load (internal/modules/cjs/loader.js:708:14)
[Appium] Welcome to Appium v2.0.0-beta.13 (REV 971920321bcd9b649a37580c3802a94834e2c955)
[Appium] Non-default server args:
[Appium]   plugins: {
[Appium]     0: images
[Appium]   }
[Appium]   tmpDir: /var/folders/jt/k0t655c90bqbtl27zfww65fh0000gn/T
[Appium] Attempting to load driver uiautomator2...
[Appium] Appium REST http interface listener started on 0.0.0.0:4723
[Appium] Available drivers:
[Appium]   - uiautomator2@1.67.0 (automationName 'UiAutomator2')
[Appium] Available plugins:
[Appium]   - images@1.2.1
[Appium] No plugins activated. Use the --plugins flag with names of plugins to activate

**Comment by jlipps** [comment, marked as answer]
This is a bug, can you create an issue instead?

**Reply by jlipps** [reply]
I think I've fixed it, if you run `appium plugin update images` to get the latest, it should work now.

**Reply by RSM-SBA** [reply]
Thank you very much @jlipps for this quick fix ! You saved my day :)
Tested it a few minutes ago and the server starts now correctly and activates the plugin.

**Reply by jlipps** [reply]
Great! Next time don't wait a _whole_ day before asking for help. With new stuff like this it could very well be a bug :-)

**Reply by sakibguy** [reply]
Thanks https://github.com/appium/appium/discussions/15818#discussioncomment-1302466

**Reply by shraeyakbasu** [reply]
Hi 
I am facing issue to start the server using desired-caps plugin 
Installed appium using - npm install -g appium@next
Installed driver using - appium driver install uiautomator2 
Installed plugin using - appium plugin install desired-caps
Activating the plugin using - appium --use-plugins desired-caps 

*** I am not able to activate relaxed-caps plugin but could activate images and execute-driver plugin 
............

Microsoft Windows [Version 10.0.19044.1706]
(c) Microsoft Corporation. All rights reserved.

C:\Users\Santosh>appium plugin list
√ Listing available plugins
- images@1.1.10 [installed (NPM)]
- execute-driver@1.0.5 [installed (NPM)]
- relaxed-caps@1.0.0-beta.6 [installed (NPM)]

C:\Users\Santosh>appium --use-plugins relaxed-caps
[Appium] Attempting to load plugin relaxed-caps...
[debug] [Appium] Requiring plugin at C:\Users\Santosh\.appium\node_modules\@appium\relaxed-caps-plugin
[Appium] Could not load plugin 'relaxed-caps', so it will not be available. Error in loading the plugin was: Cannot set property 'pluginName' of undefined
[debug] [Appium] TypeError: Cannot set property 'pluginName' of undefined
[debug] [Appium]     at map (C:\Users\Santosh\AppData\Roaming\npm\node_modules\appium\lib\extension\index.js:55:11)
[debug] [Appium]     at Array.map (<anonymous>)
[debug] [Appium]     at getActivePlugins (C:\Users\Santosh\AppData\Roaming\npm\node_modules\appium\lib\extension\index.js:50:8)
[debug] [Appium]     at main (C:\Users\Santosh\AppData\Roaming\npm\node_modules\appium\lib\main.js:263:25)
[Appium] Welcome to Appium v2.0.0-beta.35
[Appium] Non-default server args:
[Appium] {
[Appium]   usePlugins: [
[Appium]     'relaxed-caps'
[Appium]   ]
[Appium] }
[Appium] Attempting to load driver uiautomator2...
[debug] [Appium] Requiring driver at C:\Users\Santosh\.appium\node_modules\appium-uiautomator2-driver
[Appium] Attempting to load driver windows...
[debug] [Appium] Requiring driver at C:\Users\Santosh\.appium\node_modules\appium-windows-driver
[Appium] Attempting to load driver xcuitest...
[debug] [Appium] Requiring driver at C:\Users\Santosh\.appium\node_modules\appium-xcuitest-driver
[Appium] Attempting to load driver YouiEngine...
[debug] [Appium] Requiring driver at C:\Users\Santosh\.appium\node_modules\appium-youiengine-driver
[Appium] Attempting to load driver espresso...
[debug] [Appium] Requiring driver at C:\Users\Santosh\.appium\node_modules\appium-espresso-driver
(node:4524) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 exit listeners added to [process]. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)
[Appium] Appium REST http interface listener started on 0.0.0.0:4723
[Appium] Available drivers:
[Appium]   - uiautomator2@2.1.3 (automationName 'UiAutomator2')
[Appium]   - windows@2.0.5 (automationName 'Windows')
[Appium]   - xcuitest@4.4.2 (automationName 'XCUITest')
[Appium]   - YouiEngine@1.2.9 (automationName 'YouiEngine')
[Appium]   - espresso@2.3.0 (automationName 'Espresso')
[Appium] Available plugins:
[Appium]   - images@1.1.10
[Appium]   - execute-driver@1.0.5
[Appium]   - relaxed-caps@1.0.0-beta.6
[Appium] No plugins activated. Use the --use-plugins flag with names of plugins to activate
[Appium] Received SIGINT - shutting down
[debug] [AppiumDriver@f086] There are no active sessions for cleanup
[HTTP] Waiting until the server is closed
[HTTP] Received server close event
Terminate batch job (Y/N)? y

C:\Users\Santosh>appium --use-plugins images
[Appium] Attempting to load plugin images...
[debug] [Appium] Requiring plugin at C:\Users\Santosh\.appium\node_modules\@appium\images-plugin
[Appium] Welcome to Appium v2.0.0-beta.35
[Appium] Non-default server args:
[Appium] {
[Appium]   usePlugins: [
[Appium]     'images'
[Appium]   ]
[Appium] }
[Appium] Attempting to load driver uiautomator2...
[debug] [Appium] Requiring driver at C:\Users\Santosh\.appium\node_modules\appium-uiautomator2-driver
[Appium] Attempting to load driver windows...
[debug] [Appium] Requiring driver at C:\Users\Santosh\.appium\node_modules\appium-windows-driver
[Appium] Attempting to load driver xcuitest...
[debug] [Appium] Requiring driver at C:\Users\Santosh\.appium\node_modules\appium-xcuitest-driver
[Appium] Attempting to load driver YouiEngine...
[debug] [Appium] Requiring driver at C:\Users\Santosh\.appium\node_modules\appium-youiengine-driver
[Appium] Attempting to load driver espresso...
[debug] [Appium] Requiring driver at C:\Users\Santosh\.appium\node_modules\appium-espresso-driver
(node:10180) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 exit listeners added to [process]. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)
[Appium] Appium REST http interface listener started on 0.0.0.0:4723
[Appium] Available drivers:
[Appium]   - uiautomator2@2.1.3 (automationName 'UiAutomator2')
[Appium]   - windows@2.0.5 (automationName 'Windows')
[Appium]   - xcuitest@4.4.2 (automationName 'XCUITest')
[Appium]   - YouiEngine@1.2.9 (automationName 'YouiEngine')
[Appium]   - espresso@2.3.0 (automationName 'Espresso')
[Appium] Available plugins:
[Appium]   - images@1.1.10 (ACTIVE)
[Appium]   - execute-driver@1.0.5
[Appium]   - relaxed-caps@1.0.0-beta.6

**Reply by jlipps** [reply]
@shraeyakbasu this is a bug report, please create an issue instead of spamming an old discussion

**Reply by jlipps** [reply]
fyi @boneskull i was able to reproduce this, seems like an error in extension loading?

**Reply by boneskull** [reply]
Please see PR #16983 for a fix.
